data/helpers.d/package

```diff
@@ -55,48 +71,72 @@ ynh_package_install() {
         || ynh_package_install equivs
 
     # retrieve package information
-    pkgname=$(grep '^Package: ' $controlfile | cut -d' ' -f 2)
-    pkgversion=$(grep '^Version: ' $controlfile | cut -d' ' -f 2)
+    pkgname=$(grep '^Package: ' $controlfile | cut -d' ' -f 2)	# Retrieve the name of the debian package
+    pkgversion=$(grep '^Version: ' $controlfile | cut -d' ' -f 2)	# And its version number
     [[ -z "$pkgname" || -z "$pkgversion" ]] \
-        && echo "Invalid control file" && exit 1
+        && echo "Invalid control file" && exit 1	# Check if this 2 variables aren't empty.
 
-    # update packages cache
+    # Update packages cache
     ynh_package_update
 
-    # build and install the package
-    TMPDIR=$(ynh_mkdir_tmp)
-    (cp "$controlfile" "${TMPDIR}/control" \
-     && cd "$TMPDIR" \
+    # Build and install the package
+    TMPDIR=$(mktemp -d)
+    # Note that the cd executes into a sub shell
+    # Create a fake deb package with equivs-build and the given control file
+    # Install the fake package without its dependencies with dpkg
+    # Install missing dependencies with ynh_package_install
+    (cp "$controlfile" "${TMPDIR}/control" && cd "$TMPDIR" \
      && equivs-build ./control 1>/dev/null \
      && sudo dpkg --force-depends \
           -i "./${pkgname}_${pkgversion}_all.deb" 2>&1 \
-     && ynh_package_install -f) \
-        && ([[ -n "$TMPDIR" ]] && rm -rf $TMPDIR)
+     && ynh_package_install -f)
+    [[ -n "$TMPDIR" ]] && rm -rf $TMPDIR	# Remove the temp dir.
 
     # check if the package is actually installed
     ynh_package_is_installed "$pkgname"
 }
 
-# Remove package(s)
+# Install dependencies with a equivs control file
 #
-# usage: ynh_package_remove name [name [...]]
-# | arg: name - the package name to remove
-ynh_package_remove() {
-    ynh_apt remove $@
+# usage: ynh_app_dependencies dep [dep [...]]
+# | arg: dep - the package name to install in dependence
+ynh_app_dependencies () {
+    dependencies=$@
+    manifest_path="../manifest.json"
+    if [ ! -e "$manifest_path" ]; then
+    	manifest_path="../settings/manifest.json"	# Into the restore script, the manifest is not at the same place
+    fi
+    version=$(sudo python3 -c "import sys, json;print(json.load(open(\"$manifest_path\"))['version'])")	# Retrieve the version number in the manifest file.
+    dep_app=${app//_/-}	# Replace all '_' by '-'
+    cat > ./${dep_app}-ynh-deps.control << EOF	# Make a control file for equivs-build
+Section: misc
+Priority: optional
+Package: ${dep_app}-ynh-deps
+Version: ${version}
+Depends: ${dependencies// /, }
+Architecture: all
+Description: Fake package for ${app} (YunoHost app) dependencies
+ This meta-package is only responsible of installing its dependencies.
+EOF
+    ynh_package_install_from_equivs ./${dep_app}-ynh-deps.control \
+        || ynh_die "Unable to install dependencies"	# Install the fake package and its dependencies
+    ynh_app_setting_set $app apt_dependencies $dependencies
```

Comment:
So, do I understand this correctly : this helper (ynh_app_dependencies) should be called once and only once in the install script, right ? Otherwise this helper will overwrite the equivs control file and this line (ynh_app_setting_set) will overwrite the previous apt_dependencies.

Is this expected ? I ask this specifically about the Abiword/etherpad thing you were talking about. Are you preparing a list of dependencies THEN calling this helper with the entire list - or are you calling this helper two times, once with the mandatory depencies and once with the optional dependencies ?